Assets/__Scripts/Map/Notes/BeatmapNoteContainer.cs

```diff
@@ -124,8 +128,11 @@ public void SetIsPlaying(bool isPlaying)
          * 
          * So hard, in fact, that I've given up trying, and instead moved to storing references shaders and swapping them out.
          */
-        Material material = ModelMaterials[0];
-        material.shader = isPlaying ? transparentShader : opaqueShader;
-        material.SetFloat("_Editor_IsPlaying", isPlaying ? 1 : 0);
+        for (int i = 0; i < 4; i++) // Magic number is bad
```

Comment:
Magic number _is_ bad! Have you considered using `ModelMaterials.Count`?

Reply:
lol
I remember making this comment to make myself go back and fix this and never did, I think it's more complicated than that but I'll get back to you